src/javascript/campaign-url-builder/components/campaign-url.js

```diff
@@ -169,44 +208,71 @@ export default class CampaignUrl extends React.Component {
               </label>
             </div>
           </div>
-          {this.renderActionsButton()}
+          {problematicElement}
+          {this.renderActionsButton(problematicElement ? true : false)}
           <div ref="url" className="u-visuallyHidden">{url}</div>
         </div>
       </div>
     );
   }
 
+  // NOTE(nathanwest): In the renderActionsButton, below, the IconButton
+  // elements all have keys. This is because many of them are conditionally
+  // rendered, which can confuse React's renderer, because it thinks that
+  // a button that just disappeared and a button that just appeared are
+  // all the same button. The keys let react know that they are different
+  // buttons, which ensures that things like button state
+  // (idle/hover/active/pressed) aren't eroneously preserved between
+  // different buttons.
 
   /**
    * Renders the Copy to Clipboard and Shorten URL buttons.
-   * @return {Object}
+   * @param {boolean} isProblematic If this parameter is true, the buttons
+   * are hidden behind an "I know what I'm doing" warning button.
+   * @return {JSX Literal}
    */
-  renderActionsButton() {
+  renderActionsButton(isProblematic) {
     return supports.copyToClipboard() ? (
```

Comment:
This ternary is hard to parse, could you pull out the separate paths into named functions?